curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/644fa585a5f9970173ca3cff.md

```diff
@@ -0,0 +1,348 @@
+  justify-content: center;
+  gap: 10px;
+  flex-wrap: wrap;
+}
+
+.avatar-container img {
+  width: 30px;
+  height: 30px;
+}
+
+@media (max-width: 750px) {
+  .table-wrapper {
+    padding: 0 15px;
+  }
+
+  table {
+    width: 700px;
+  }
+
+  th {
+    font-size: 1.2rem;
+  }
+
+  .post-title {
+    font-size: 1.1rem;
+  }
+}
+```
+
+```js
+const FORUM_LATEST = "https://forum-proxy.freecodecamp.rocks/latest";
+const FORUM_TOPIC_URL = "https://forum.freecodecamp.org/t/";
+const FORUM_CATEGORY_URL = "https://forum.freecodecamp.org/c/";
+const AVATAR_URL = "https://sea1.discourse-cdn.com/freecodecamp";
+
+const postsContainer = document.getElementById("posts-container");
+
+const allCategories = {
+  299: { category: "Career Advice", className: "career" },
+  409: { category: "Project Feedback", className: "feedback" },
+  417: { category: "freeCodeCamp Support", className: "support" },
+  421: { category: "JavaScript", className: "javascript" },
+  423: { category: "HTML - CSS", className: "html-css" },
+  424: { category: "Python", className: "python" },
+  432: { category: "You Can Do This!", className: "motivation" },
+  560: { category: "Backend Development", className: "backend" },
+};
+
+const forumCategory = (id) => {
+  let selectedCategory = {};
+
+  if (allCategories.hasOwnProperty(id)) {
+    const { className, category } = allCategories[id];
+
+    selectedCategory.className = className;
+    selectedCategory.category = category;
+  } else {
+    selectedCategory.className = "General";
+    selectedCategory.category = "General";
+    selectedCategory.id = 1;
+  }
+  const url = `${FORUM_CATEGORY_URL}${selectedCategory.className}/${id}`;
+  const linkText = selectedCategory.category;
+--fcc-editable-region--
+
+--fcc-editable-region--
+};
+
+const timeAgo = (time) => {
+  const currentTime = new Date();
+  const lastPost = new Date(time);
+
+  const timeDifference = currentTime - lastPost;
+  const MS_PER_MINUTE = 1000 * 60;
+
+  const minutesAgo = Math.floor(timeDifference / MS_PER_MINUTE);
+  const hoursAgo = Math.floor(minutesAgo / 60);
+  const daysAgo = Math.floor(hoursAgo / 24);
+
+  if (minutesAgo < 60) {
+    return `${minutesAgo}m ago`;
+  }
+
+  if (hoursAgo < 24) {
+    return `${hoursAgo}h ago`;
+  }
+
+  return `${daysAgo}d ago`;
+};
+
+const viewCount = (views) => {
+  const thousands = Math.floor(views / 1000);
+
+  if (views >= 1000) {
+    return `${thousands}k`;
+  }
+
+  return views;
+};
+
+const fetchData = async () => {
+  try {
+    const res = await fetch(FORUM_LATEST);
+    const data = await res.json();
+    console.log(data);
+    showLatestPosts(data);
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+fetchData();
+
+const showLatestPosts = (data) => {
+  const { topic_list, users } = data;
+  const { topics } = topic_list;
+
+  postsContainer.innerHTML = topics.map((item) => {
+    const {
+      id,
+      title,
+      views,
+      posts_count,
+      slug,
+      posters,t,
+    } = item;
```

Comment:
This has some errors in it and as a result is breaking the layout here

<img width="615" alt="Screenshot 2023-11-21 at 4 34 42 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/b4c6e1ce-d08d-4871-ae2f-d9075a9c1a76">

```suggestion
    const {
      id,
      title,
      views,
      posts_count,
      slug,
      posters,
      category_id,
      bumped_at,
    } = item;
```